sopel/loader.py

```diff
@@ -110,12 +110,16 @@ def enumerate_modules(config, show_all=False):
     return modules
 
 
-def compile_rule(nick, pattern):
+def compile_rule(nick, pattern, alias_nicks):
     # Not sure why this happens on reloads, but it shouldn't cause problems…
     if isinstance(pattern, _regex_type):
         return pattern
 
     nick = re.escape(nick)
+    if alias_nicks:
+        nicks = alias_nicks
+        nicks.append(nick)
+        nick = '(%s)' % '|'.join(nicks)
```

Comment:
Just realized a situation where this could fail, if one of the alias nicks contains a pipe or any other regex special character, it may parse the nick differently. For example, if someone were to add `sopel|gone` as an alias nick, it wouldn't match `sopel|gone` but `sopel` and `gone` individually.